src/mixins/nft.js

```diff
@@ -715,9 +716,34 @@ export default {
         actionType,
         ignoreToList,
       });
-      this.NFTHistory = this.nftIsWritingNFT
-        ? populateGrantEvent(latestBatchEvents, dbEventMap)
-        : latestBatchEvents;
+
+      const nftBookLatestBatchEvents = latestBatchEvents;
+      if (this.nftIsNFTBook) {
+        const { messages: nftBookBuyerMessages } = await this.$api.$get(
+          getNftBookBuyerMessage(this.classId)
+        );
+        for (let i = 0; i < nftBookBuyerMessages.length; i += 1) {
+          const buyerMessage = nftBookBuyerMessages[i].message;
+          const { txHash } = nftBookBuyerMessages[i];
+
+          const matchingEvent = nftBookLatestBatchEvents.find(
+            event => event.txHash === txHash
+          );
+
+          if (matchingEvent) {
+            matchingEvent.buyerMessage = buyerMessage;
+          }
+        }
+      }
+
+      if (this.nftIsWritingNFT) {
+        this.NFTHistory = populateGrantEvent(latestBatchEvents, dbEventMap);
+      } else if (this.nftIsNFTBook) {
+        this.NFTHistory = nftBookLatestBatchEvents;
+      } else {
+        this.NFTHistory = latestBatchEvents;
```

Comment:
I think `nftBookLatestBatchEvents` and `latestBatchEvents` are same since array is pass by ref, no need `if else`